Tests/BaseLib/TestgetParenthesizedString.cpp

```diff
@@ -0,0 +1,138 @@
+        ASSERT_TRUE(std::get<0>(BaseLib::getParenthesizedString(
+                                    test_string, '{', '}', pre.length() + 2))
+                        .empty());
+    }
+
+    // test "some_random_string{some_random_string}some_random_string"
+    for (int i = 0; i < 100; ++i)
+    {
+        std::string const pre = generate_random_alphanumeric_string(20);
+        std::string const post = generate_random_alphanumeric_string(20);
+        std::string const expected = generate_random_alphanumeric_string(20);
+        std::string const test_string = pre + "{" + expected + "}" + post;
+        ASSERT_EQ(expected,
+                  std::get<0>(BaseLib::getParenthesizedString(test_string, '{',
+                                                              '}', 0)));
+        ASSERT_TRUE(std::get<0>(
+            BaseLib::getParenthesizedString(
+                test_string, '{', '}', pre.length() + expected.length() + 2))
+                .empty());
+    }
+
+    // test "some_random_string{some_random_string{some_random_string}"
+    for (int i = 0; i < 100; ++i)
+    {
+        std::string const pre = generate_random_alphanumeric_string(20);
+        std::string const post = generate_random_alphanumeric_string(20);
+        std::string const expected =
+            generate_random_alphanumeric_string(20) + "{" + post;
+        std::string const test_string = pre + "{" + expected + "}";
+        ASSERT_EQ(expected,
+                  std::get<0>(BaseLib::getParenthesizedString(test_string, '{',
+                                                              '}', 0)));
+        ASSERT_TRUE(std::get<0>(BaseLib::getParenthesizedString(
+                                    test_string, '{', '}',
+                                    test_string.length() - post.length() + 1))
+                        .empty());
+    }
+
+    // test "some_random_string{some_random_string}{some_random_string}"
+    for (int i = 0; i < 100; ++i)
+    {
+        std::string const pre = generate_random_alphanumeric_string(20);
+        std::string const expected_1 = generate_random_alphanumeric_string(20);
+        std::string const expected_2 = generate_random_alphanumeric_string(20);
+        std::string const test_string =
+            pre + "{" + expected_1 + "}{" + expected_2 + "}";
+        ASSERT_EQ(expected_1,
+                  std::get<0>(BaseLib::getParenthesizedString(test_string, '{',
+                                                              '}', 0)));
+        ASSERT_EQ(
+            expected_2,
+            std::get<0>(BaseLib::getParenthesizedString(
+                test_string, '{', '}', pre.length() + expected_1.length())));
+        ASSERT_TRUE(
+            std::get<0>(BaseLib::getParenthesizedString(
+                            test_string, '{', '}',
+                            test_string.length() - expected_2.length() + 1))
+                .empty());
+    }
```

Comment:
Maybe a few more tests for different positions of curly braces would be good:
 - {}
 - }a{
 - }a{b}
 - {a{b}

Reply:
More tests are added.